src/promptflow-devkit/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -134,18 +135,33 @@ def _try_write_trace_to_cosmosdb(
         )
         collection_client_thread.start()
 
+        line_summary_client_thread = ThreadWithContextVars(
+            target=get_client,
+            args=(CosmosDBContainerName.LINE_SUMMARY, subscription_id, resource_group_name, workspace_name, credential),
+        )
+        line_summary_client_thread.start()
+
         # Load created_by info first time may slow. So, we load it in parallel for warm up.
         created_by_thread = ThreadWithContextVars(target=get_created_by_info_with_cache)
         created_by_thread.start()
 
-        get_client(CosmosDBContainerName.LINE_SUMMARY, subscription_id, resource_group_name, workspace_name, credential)
+        # Get default blob may be slow. So, we have a cache for default datastore.
+        from promptflow.azure._storage.blob.client import get_datastore_container_client
```

Comment:
Is there any particular reason to import inside a function? It’s more of a style question, and the decision is entirely up to you. 😆

Reply:
If not enable local to cloud trace, we should not import azure related lib here.

Reply:
Which means `_try_write_trace_to_cosmosdb` should not be called at all?

Reply:
Yes. But for current implementation, we need to check the span data to decide if the cloud trace is enable. And we may not want such logic appear in upper level handle.